src/Reflection/ClassReflection.php

```diff
@@ -113,65 +108,55 @@ public function getMethod($name)
     }
 
     /**
-     * Get reflection objects of all methods
+     * {@inheritDoc}
      *
      * @param  int $filter
-     * @return MethodReflection[]
+     * @return list<MethodReflection>
      */
     #[ReturnTypeWillChange]
     public function getMethods($filter = -1)
     {
-        $methods = [];
-        foreach (parent::getMethods($filter) as $method) {
-            $instance  = new MethodReflection($this->getName(), $method->getName());
-            $methods[] = $instance;
-        }
+        $name = $this->getName();
 
-        return $methods;
+        return array_map(
+            static fn (ReflectionMethod $method): MethodReflection => new MethodReflection($name, $method->getName()),
+            parent::getMethods($filter)
+        );
     }
 
     /**
-     * Returns an array of reflection classes of traits used by this class.
+     * {@inheritDoc}
      *
-     * @return null|array
+     * @return array<trait-string, ClassReflection>
      */
     #[ReturnTypeWillChange]
     public function getTraits()
     {
-        $vals   = [];
-        $traits = parent::getTraits();
-        if ($traits === null) {
-            return;
-        }
-
-        foreach ($traits as $trait) {
-            $vals[] = new ClassReflection($trait->getName());
-        }
-
-        return $vals;
+        return array_map(
+            static fn (ReflectionClass $trait): ClassReflection => new ClassReflection($trait->getName()),
+            parent::getTraits()
```

Comment:
This isn't marked internal, so we may need to still return `null` instead of an empty array if `parent::getTraits()` returns null.

Reply:
TBH, that never happens: https://3v4l.org/cl2aI

Also not nullable in `roave/better-reflection`, where we improved on this signature quite a bit: https://github.com/Roave/BetterReflection/blob/631d23c6b2226058f3a96c7ab21658ea2d05df64/src/Reflection/Adapter/ReflectionClass.php#L319-L331